elastic/kibana
# [Upgrade Assistant] Fix isSuccessful check in verifySnapshotUpgrade method

# Summary

When we attempt to update a ML snapshot model we have a method thast checks if the upgrade was successful. If we have only one deprecation the [first condition](https://github.com/elastic/kibana/blob/7.17/x-pack/plugins/upgrade_assistant/server/routes/ml_snapshots.ts#L73) from this method will succeed and it will return that the snapshot was migrated correctly. But if we have more than one snapshot to be migrated this condition doesn't run and instead it will [attempt to find](https://github.com/elastic/kibana/blob/7.17/x-pack/plugins/upgrade_assistant/server/routes/ml_snapshots.ts#L79) in the `mlSnapshotDeprecations` list if we have a match for the snapshot we're updating and nothing is found then the update was successful.

The problem was that there was some alterations made to the format of the deprecation titles and now it has 3 sets of brackets instead of two, so the UI was reporting that the upgrade failed while in reality it succeeded.

**Steps to test:**
1. Run elasticsearch and kibana
2. Create and run **TWO** really simple ML anomaly detection jobs. This can be run on Kibana sample data, for example the sample flight data, and can be as simple as possible, for example a single metric count job.  Just run the job on the historical data, don't ask for it to continue in real-time.
3. Go to dev tools and update the model snapshot definition docs to say they came from 6.3:

```
GET .ml-anomalies-shared/_search
{
  "query": {
    "exists": { "field" : "snapshot_id" }
  }
}

POST .ml-anomalies-shared/_update_by_query
{
  "query": {
    "exists": { "field" : "snapshot_id" }
  },
  "script": {
    "source": "ctx._source.min_version='6.3.0'",
    "lang": "painless"
  }
}

GET .ml-anomalies-shared/_search
{
  "query": {
    "exists": { "field" : "snapshot_id" }
  }
}
```

4. Now go to upgrade assistant and look at the deprecation warnings.  The hack in step 4 should convince it that the model snapshot created by the simple job created in step 3 needs upgrading.
5. Make sure that both ML deprecations can be addressed correctly.
6. To iterate, steps 4 and 5 can be repeated as many times as it takes.
